crates/router/src/core/payments/operations/payment_session.rs

```diff
@@ -310,131 +308,94 @@ where
+                                    api::GetToken::from(payment_method_type.payment_method_type),
+                                );
+                                connector_and_supporting_wallet.push(connector_and_wallet);
                             }
                         }
-                    })
-            })
-            .map(|filtered_connector| filtered_connector.connector_name.clone())
-            .collect();
-
-        // Parse the payment methods enabled to check if the merchant has enabled googlepay ( wallet ) using that connector.
-        // A single connector can support creating session token from metadata as well as by calling the connector.
-        let session_token_from_metadata_connectors = connector_accounts
-            .iter()
-            .filter(|connector_account| {
-                connector_account
-                    .payment_methods_enabled
-                    .clone()
-                    .unwrap_or_default()
-                    .iter()
-                    .any(|payment_method| {
-                        let parsed_payment_method_result: Result<
-                            PaymentMethodsEnabled,
-                            error_stack::Report<errors::ParsingError>,
-                        > = payment_method.clone().parse_value("payment_method");
-
-                        match parsed_payment_method_result {
-                            Ok(parsed_payment_method) => parsed_payment_method
-                                .payment_method_types
-                                .map(|payment_method_types| {
-                                    payment_method_types.iter().any(|payment_method_type| {
-                                        matches!(
-                                            payment_method_type.payment_method_type,
-                                            api_models::enums::PaymentMethodType::GooglePay
-                                        )
-                                    })
-                                })
-                                .unwrap_or(false),
-                            Err(parsing_error) => {
-                                logger::debug!(session_token_parsing_error=?parsing_error);
-                                false
-                            }
+                    }
+                    Err(parsing_error) => {
+                        logger::debug!(session_token_parsing_error=?parsing_error);
+                    }
+                }
+            }
+        }
+
+        let payment_method_types = request.wallets.clone();
+
+        let connectors_data = if !payment_method_types.is_empty() {
+            let mut connectors_data = Vec::new();
+            for payment_method_type in payment_method_types {
+                for connector_tokeninfo in &connector_and_supporting_wallet {
+                    let connector_name = match payment_method_type {
+                        api_models::enums::PaymentMethodType::Paypal => Ok("braintree"),
```

Comment:
The connector names should not be hardcoded. filter the `connector_and_supporting_wallet` array for given wallets.